File: XamarinStore.iOS/Views/AutoCompleteTextEntry.cs
using System;
using MonoTouch.UIKit;
using MonoTouch.Foundation;
using System.Collections.Generic;
using System.Linq;
using System.Drawing;

namespace XamarinStore
{
	public class AutoCompleteTextEntry : TextEntryView
	{
		StringTableViewController controller;
		public string Title { get; set; }
		public AutoCompleteTextEntry ()
		{
			controller = new StringTableViewController () {
				ItemSelected = (item) => {
					Value = item;
				}
			};
			

			textField.Started += (object sender, EventArgs e) => {
				Search ();
			};
		}

		IEnumerable<string> items = new List<string>();

		public IEnumerable<string> Items {
			get {
				return items;
			}
			set {
				items = value;
			}
		}

		void Search ()
		{
			if (items.Count() == 0)
				return;

			textField.ResignFirstResponder();
			controller.Title = Title;
			controller.Items = items;
			if(PresenterView.NavigationController.TopViewController != controller)
				PresenterView.NavigationController.PushViewController (controller, true);


		}

		public UIViewController PresenterView {get;set;}
	}
}

File: F#/XamarinStore.iOS/Views/AutoCompleteTextEntry.fs
namespace XamarinStore

open System
open MonoTouch.UIKit
open MonoTouch.Foundation
open System.Collections.Generic
open System.Linq
open System.Drawing

type AutoCompleteTextEntry() as this =
    inherit TextEntryView()

    let controller = new StringTableViewController ( ItemSelected = fun item -> this.Value <- item)
    do this.TextField.Started.Add(fun _ ->this.Search ())

    member val Items:IEnumerable<string> = new List<string>() :> IEnumerable<string> with get,set

    member this.Search () =
        if this.Items.Count() <> 0 then
            this.TextField.ResignFirstResponder() |> ignore
            controller.Title <- this.Title
            controller.Items <- this.Items.ToList()
            if this.PresenterView.NavigationController.TopViewController <> (controller :> UIViewController) then
                this.PresenterView.NavigationController.PushViewController(controller, true)

    member val PresenterView:UITableViewController = null with get,set
    member val Title:string = null with get,set
